packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Watch,
+  Prop,
+  State,
+  h,
+} from '@stencil/core';
+import { assignLanguage, observerConfig } from '../../utils/utils';
+import { inheritAttributes } from '../../utils/utils';
+import i18n from './i18n/i18n';
+
+@Component({
+  tag: 'gcds-link',
+  styleUrl: 'gcds-link.css',
+  shadow: true,
+})
+export class GcdsLink {
+  @Element() el: HTMLElement;
+
+  private shadowElement?: HTMLElement;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Set the link size
+   */
+  @Prop({ mutable: true }) size: 'regular' | 'small' = 'regular';
+
+  @Watch('size')
+  validateSize(newValue: string) {
+    const values = ['regular', 'small'];
+
+    if (!values.includes(newValue)) {
+      this.size = 'regular';
+    }
+  }
+
+  // TODO: Do we need to validate display, rel, target, etc?
+
+  private handleClick = (e: Event) => {
```

Comment:
Can this function be moved out of the props section and at the bottom of the `Events` section?